### PR TITLE
Add user profile patch route

### PR DESCRIPTION
**Pull request by AlexisDougherty13**
Added a patch route for editing user profile. Was unable to test to be sure it works but I think it will. Definitely compiles. I'll clean it up a bit when I'm able to test it for sure.

### CODE REVIEW

**Review by chris-brugal — APPROVED** [review]
looks good

File: backend/src/router.go [hunk 1]
```diff
@@ -349,3 +351,48 @@ func getProfile(c *gin.Context) {
...
+		return
+	}
+	
+	var users []User
+	//add error checking for nonexisting user 
+	//Note for self: https://blog.logrocket.com/how-to-build-a-rest-api-with-golang-using-gin-and-gorm/
+	
+	var edits ProfileUpdateData
+	if err := c.ShouldBindJSON(&edits); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
```
Comment (WillBAnders) [inline backend/src/router.go]: Minor nitpick - for consistency, let's stick to directly using `400` as the status rather than the constant. We can open an issue to make this change to all of the routes later; didn't realize this was a thing.

File: backend/src/router.go [hunk 2]
```diff
@@ -349,3 +351,48 @@ func getProfile(c *gin.Context) {
...
+	
+	claims, err := ParseJWT(token)
+	if err != nil {
+		c.JSON(500, gin.H{
+			"error": "Unable to parse JWT: " + err.Error() + ".",
+		})
+		return
+	}
+	
+	var users []User
```
Comment (WillBAnders) [inline backend/src/router.go]: Move declaration to directly before the `Find` call.

File: backend/src/router.go [hunk 3]
```diff
@@ -349,3 +351,48 @@ func getProfile(c *gin.Context) {
...
+	//Note for self: https://blog.logrocket.com/how-to-build-a-rest-api-with-golang-using-gin-and-gorm/
+	
+	var edits ProfileUpdateData
+	if err := c.ShouldBindJSON(&edits); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	
+	DB.Find(&users, "username = ?", claims.Username)
+	DB.Model(&users[0]).Updates(edits)
```
Comment (WillBAnders) [inline backend/src/router.go]: I saw the latest commit regarding some issue with Gorm here and took a closer look. [According to the docs](https://gorm.io/docs/update.html#Updates-multiple-columns), `Updates` with structs will not change default values which is what we need (that way a user can remove an email or something). We also need to be careful about distinguishing between default values passed in the body to be updated versus default values applied from binding. Not sure the best way to handle this; it's probably a separate issue.

File: backend/src/router.go [hunk 4]
```diff
@@ -349,3 +351,48 @@ func getProfile(c *gin.Context) {
...
+//not all of this is needed to make an update, at least according to the guide. We shall see.
+type ProfileUpdateData struct { 
+	ID        uint   `json:"-"`
+	FirstName string `json:"firstname"`
+	LastName  string `json:"lastname"`
+	Email     string `json:"email"`
+	Phone     string `json:"phone"`
+}
+
+// ADD DESCRIPTION
```
Comment (WillBAnders) [inline backend/src/router.go]: The route documentation should be done in this commit, that way Chris knows the intended schema for the route so he can use it on the frontend.

**Review by WillBAnders — CHANGES_REQUESTED** [review]
Only change I'm going to request is adding in the documentation for the route; it should be done here rather than in a separate PR. The other styling changes would be nice too.

The handling of update seems to be a bit of a problem, you're welcome to tackle that here or we can make an issue for it an handle it in a later PR - just make sure to note this limitation in the doc comment.

File: backend/src/router.go (hunk 1 above)
Comment (AlexisDougherty13) [reply backend/src/router.go]: Fixed to a 400 and added the documentation. I had planned on adding it soon, just wanted to get the init stuff working before I made docs and then would need to make major changes to it.

File: backend/src/router.go (hunk 2 above)
Comment (AlexisDougherty13) [reply backend/src/router.go]: Done :-)

File: backend/src/router.go (hunk 4 above)
Comment (AlexisDougherty13) [reply backend/src/router.go]: Done, although it's in progress due to adding tutors

**Review by WillBAnders — APPROVED** [review]

**Comment by WillBAnders** [comment]
Rebase and should be good to merge. We'll handle backend tests for it as part of finalizing the specification for default values (see https://github.com/WillBAnders/TutorsVILLE/pull/97#discussion_r837852527).

**Comment by WillBAnders** [comment]
Split off later commits for other frontend work into a separate PR. The route implementation still needs some work but we can address the edge cases in combination with implementing tests.
